// @google/gemini-cli/dist/src/commands/extensions/enable.js
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */
import {} from 'yargs';
import { debugLogger, FatalConfigError, getErrorMessage, } from '@google/gemini-cli-core';
import { enableExtension } from '../../config/extension.js';
import { SettingScope } from '../../config/settings.js';
import { ExtensionEnablementManager } from '../../config/extensions/extensionEnablement.js';
export function handleEnable(args) {
    const extensionEnablementManager = new ExtensionEnablementManager();
    try {
        if (args.scope?.toLowerCase() === 'workspace') {
            enableExtension(args.name, SettingScope.Workspace, extensionEnablementManager);
        }
        else {
            enableExtension(args.name, SettingScope.User, extensionEnablementManager);
        }
        if (args.scope) {
            debugLogger.log(`Extension "${args.name}" successfully enabled for scope "${args.scope}".`);
        }
        else {
            debugLogger.log(`Extension "${args.name}" successfully enabled in all scopes.`);
        }
    }
    catch (error) {
        throw new FatalConfigError(getErrorMessage(error));
    }
}
export const enableCommand = {
    command: 'enable [--scope] <name>',
    describe: 'Enables an extension.',
    builder: (yargs) => yargs
        .positional('name', {
        describe: 'The name of the extension to enable.',
        type: 'string',
    })
        .option('scope', {
        describe: 'The scope to enable the extension in. If not set, will be enabled in all scopes.',
        type: 'string',
    })
        .check((argv) => {
        if (argv.scope &&
            !Object.values(SettingScope)
                .map((s) => s.toLowerCase())
                .includes(argv.scope.toLowerCase())) {
            throw new Error(`Invalid scope: ${argv.scope}. Please use one of ${Object.values(SettingScope)
                .map((s) => s.toLowerCase())
                .join(', ')}.`);
        }
        return true;
    }),
    handler: (argv) => {
        handleEnable({
            name: argv['name'],
            scope: argv['scope'],
        });
    },
};
